python/ray/serve/_private/deployment_scheduler.py

```diff
@@ -263,10 +269,15 @@ def _get_replicas_to_stop(
         node_to_running_replicas = defaultdict(set)
         for running_replica, node_id in self._running_replicas[deployment_name].items():
             node_to_running_replicas[node_id].add(running_replica)
-        for running_replicas in sorted(
-            node_to_running_replicas.values(), key=lambda lst: len(lst)
+        # Replicas on the head node has the lowest priority for downscaling
+        # since we cannot relinquish the head node.
+        for node_and_running_replicas in sorted(
```

Comment:
```suggestion
        for node_id, running_replicas in sorted(
```
I'm not actually even sure if this is correct, it's pretty hard to follow all of the datastructure munging here